src/mbgl/annotation/shape_annotation_impl.cpp

```diff
@@ -19,121 +18,37 @@ ShapeAnnotationImpl::ShapeAnnotationImpl(const AnnotationID id_, const uint8_t m
-            converted.push_back(convertPoints(geojsonvt::ProjectedFeatureType::Polygon, ring));
-        }
-        return convertFeature(geojsonvt::ProjectedFeatureType::Polygon, converted);
-    }
-
-    geojsonvt::ProjectedFeature operator()(const MultiLineString<double>& lines) const {
-        geojsonvt::ProjectedRings converted;
-        for (const auto& line : lines) {
-            converted.push_back(convertPoints(geojsonvt::ProjectedFeatureType::LineString, line));
-        }
-        return convertFeature(geojsonvt::ProjectedFeatureType::LineString, converted);
-    }
-
-    geojsonvt::ProjectedFeature operator()(const MultiPolygon<double>& polygons) const {
-        geojsonvt::ProjectedRings converted;
-        for (const auto& polygon : polygons) {
-            for (const auto& ring : polygon) {
-                converted.push_back(convertPoints(geojsonvt::ProjectedFeatureType::Polygon, ring));
-            }
-        }
-        return convertFeature(geojsonvt::ProjectedFeatureType::Polygon, converted);
-    }
-
-private:
-    geojsonvt::LonLat convertPoint(const Point<double>& p) const {
-        return {
-            util::wrap(p.x, -util::LONGITUDE_MAX, util::LONGITUDE_MAX),
-            util::clamp(p.y, -util::LATITUDE_MAX, util::LATITUDE_MAX)
-        };
-    }
-
-    geojsonvt::ProjectedRing convertPoints(geojsonvt::ProjectedFeatureType type, const std::vector<Point<double>>& points) const {
-        std::vector<geojsonvt::LonLat> converted;
-
-        for (const auto& p : points) {
-            converted.push_back(convertPoint(p));
-        }
-
-        assert(points.size() > 0);
-        if (type == geojsonvt::ProjectedFeatureType::Polygon && points.size() > 0 && points.front() != points.back()) {
-            converted.push_back(converted.front());
-        }
-
-        return geojsonvt::Convert::projectRing(converted, tolerance);
-    }
-
-    geojsonvt::ProjectedFeature convertFeature(geojsonvt::ProjectedFeatureType type, const geojsonvt::ProjectedRings& rings) const {
-        return geojsonvt::Convert::create(geojsonvt::Tags(), type, rings);
-    }
-};
-
 void ShapeAnnotationImpl::updateTileData(const CanonicalTileID& tileID, AnnotationTileData& data) {
     static const double baseTolerance = 4;
 
     if (!shapeTiler) {
-        const uint64_t maxAmountOfTileFeatures = (1ull << maxZoom) * util::EXTENT;
-        const double tolerance = baseTolerance / maxAmountOfTileFeatures;
-
-        std::vector<geojsonvt::ProjectedFeature> features = {
-            ShapeAnnotationGeometry::visit(geometry(), ToGeoJSONVT(tolerance))
-        };
-
+        mapbox::geometry::feature_collection<double> features;
+        features.emplace_back(ShapeAnnotationGeometry::visit(geometry(), [] (auto&& geom) {
+            return Feature(std::move(geom));
+        }));
         mapbox::geojsonvt::Options options;
-        options.maxZoom = maxZoom;
+        options.maxZoom = mbgl::util::clamp((int)maxZoom, 0, 18);
```

Comment:
This looks similar to the issue I am debugging in https://github.com/mapbox/mapbox-gl-native/pull/5517. I've previously lowered the maximum zoom range for `AnnotationSource` from 22 to 18 alleviate the issue of disappearing annotations in z19 [here](https://github.com/mapbox/mapbox-gl-native/pull/5418).